{"bug_class": "PROVER", "kind": "ERROR", "bug_type": "RESOURCE_LEAK", "qualifier": "resource of type `java.io.FileReader` acquired by call to `new()` at line 99 is not released after line 116.\n**Note**: potential exception at line 102", "severity": "HIGH", "visibility": "user", "line": 116, "column": -1, "procedure": "void Cryptogram.model()", "procedure_id": "org.jacop.examples.fd.Cryptogram.model():void.69e6dcda343a28bb723dc2075429f40c", "procedure_start_line": 90, "file": "src/main/java/org/jacop/examples/fd/Cryptogram.java", "bug_trace": [{"level": 0, "filename": "src/main/java/org/jacop/examples/fd/Cryptogram.java", "line_number": 90, "column_number": -1, "description": "start of procedure model()"}, {"level": 0, "filename": "src/main/java/org/jacop/examples/fd/Cryptogram.java", "line_number": 94, "column_number": -1, "description": "Taking true branch"}, {"level": 0, "filename": "src/main/java/org/jacop/examples/fd/Cryptogram.java", "line_number": 99, "column_number": -1, "description": ""}, {"level": 0, "filename": "src/main/java/org/jacop/examples/fd/Cryptogram.java", "line_number": 102, "column_number": -1, "description": "exception java.io.IOException"}, {"level": 0, "filename": "src/main/java/org/jacop/examples/fd/Cryptogram.java", "line_number": 116, "column_number": -1, "description": ""}], "key": "Cryptogram.java|model|RESOURCE_LEAK", "node_key": "f645aafdf4c7de4e5a37ff7e8f2268a3", "hash": "a893b6c2953069e8a2332c7f7c8fb738", "bug_type_hum": "Resource Leak", "censored_reason": ""}